bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+from coalib.parsing.StringProcessing.Core import (search_for,
+                                                  search_in_between,
+                                                  unescaped_search_in_between)
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.HiddenResult import HiddenResult
+
+
+class AnnotationBear(LocalBear):
+
+    def run(self, filename, file: str, language: str, language_family: str):
+        """
+        Finds out all the positions of comments and strings in a file.
+        The Bear searches for valid comments and strings
+        and yields them as HiddenResults with the appropriate ranges.
+        Invalid comments or strings are:
+        1. Comments within comments,
+        2. Comments within strings and
+        3. Strings within comments.
+        """
+        file = ''.join(file)
```

Comment:
Never try to change `file`.
I think we may move it into a proxy later. Ideally a bear shouldn't change `file`
Can you rename this ? Maybe `text` or `lines`